backend/controllers/posts.controller.js

```diff
@@ -58,4 +58,56 @@ exports.getOtherUsersPosts = async (req, res) =>{
     } catch (error) {
         return res.status(500).send({message: `Server error: ${error.message}`})
     }
+}
+
+exports.deletePost = async (req, res) =>{
+    try{
+        const postId = req.params.id;
+        const postBy = Post.findById(postId).postBy;
+        const userId = res.locals.userId // get user id from authentication middleware
+        if(userId == postBy)
+        {
+            const result = await Post.findByIdAndRemove(postId);          
+            return res.status(200).json(result); 
+        }
+        else
+        {
+            return res.status(400).json({error: "The post is not posted by this user"});
+        }
+              
+    }catch (error) {
+    return res.status(500).send({message: `Server error: ${error.message}`})
+    }
+}
+
+exports.editPost = async (req, res) => {
+
+    try {
+        const postID = req.params.id;
+        const postBy = Post.findById(postId).postBy;
+        const userId = res.locals.userId // get user id from authentication middleware
+        if(userId == postBy)
+        {
+        const updatedPost = new Post({
```

Comment:
you also don't need to create a new Post here, just provide it as a json object, and remove id from the properties please